quickwit/quickwit-common/src/metrics.rs

```diff
@@ -19,29 +19,35 @@
 
 use prometheus::{Encoder, HistogramOpts, Opts, TextEncoder};
 pub use prometheus::{
-    Histogram, HistogramTimer, HistogramVec, IntCounter, IntCounterVec, IntGauge, IntGaugeVec,
+    Histogram, HistogramTimer, HistogramVec, IntCounter, IntCounterVec as PrometheusIntCounterVec,
+    IntGauge, IntGaugeVec,
 };
 
+pub struct IntCounterVec<const N: usize> {
+    underlying: PrometheusIntCounterVec,
+}
+
+impl<const N: usize> IntCounterVec<N> {
+    pub fn new(name: &str, description: &str, namespace: &str, labels: [&str; N]) -> Self {
+        let counter_opts = Opts::new(name, description).namespace(namespace);
```

Comment:
The constructor registers the Counter. Despite the similar naming, that's a very different behavior from the prom counters. 

Can we put the code for this constructor in a `new_counter_vec` function.

```
pub fn new_counter_vec<const N: usize>(...) -> IntCounterVec<N> {
...
}
```

Reply:
Ok.